packages/integration-sdk-runtime/src/execution/jobState.ts

```diff
@@ -275,7 +313,11 @@ export function createStepJobState({
 
     hasKey: (_key: string | undefined) => {
       if (!_key) return false;
-      return duplicateKeyTracker.hasKey(_key);
+      if (hasHasKeyFn) {
+        return graphObjectStore.hasKey!(_key);
+      } else {
+        return duplicateKeyTracker.hasKey(_key);
```

Comment:
This PR does not modify the _adding_ of a key to the `duplicateKeyTracker` only the use of the hasKey. Follow up PR to no longer add to the `duplicateKeyTracker` is coming?
I ask because we use the `duplicateKeyTracker` in the above findEntity method.